src/api/java/com/minecolonies/api/crafting/IRecipeStorageFactory.java

```diff
@@ -74,6 +76,11 @@ default RecipeStorage getNewInstance(@NotNull final IFactoryController factoryCo
             throw new IllegalArgumentException("Ninth parameter is supposed to be a ResourceLocation or Null!");
         }
 
+        if (context.length > MIN_PARAMS_IRECIPESTORAGE + 6 && context[9] != null && !(context[9] instanceof IToolType))
```

Comment:
Gotta make sure this loads old worlds well too

Reply:
Existing worlds don't use this constructor, they use the one that restores from NBT.  Then it will also build the new recipe (with tool) from the datapack and the auto-recipe code will replace the old recipe with the new one.

In testing, this just works.  Unless you can think of something I might have overlooked?